google/resource_compute_attached_disk.go

```diff
@@ -0,0 +1,227 @@
+package google
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	compute "google.golang.org/api/compute/v1"
+)
+
+func resourceComputeAttachedDisk() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAttachedDiskCreate,
+		Read:   resourceAttachedDiskRead,
+		Delete: resourceAttachedDiskDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: resourceAttachedDiskImport,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(300 * time.Second),
+			Delete: schema.DefaultTimeout(300 * time.Second),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"attached_disk": {
+				Type:             schema.TypeString,
+				Required:         true,
+				ForceNew:         true,
+				DiffSuppressFunc: compareSelfLinkOrResourceName,
+			},
+			"attached_instance": {
+				Type:             schema.TypeString,
+				Required:         true,
+				ForceNew:         true,
+				DiffSuppressFunc: compareSelfLinkOrResourceName,
+			},
+			"project": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"zone": {
```

Comment:
Similarly, we probably want the user to be able to specify this, unless disks and instances are both unique across all zones.

Reply:
I left zone & project as computed, because the `attached_disk` resource doesn't really have a project or zone of itself. The zone and project are actually more like properties of the `compute_instance` and are only used to construct the url of the compute instance. 

AFAIK the only reason somebody would need to specify the zone/project in the attached disk resource would be if they were referencing the compute instance by `name` instead of `self_link` (since I parse the self link to compute these) and they hadn't specified a global project+zone. In which case in the error I'm throwing it says "try using self link". 

Does that make sense or am I missing a valid use case for needing to override those variables?

Reply:
I think that makes perfect sense, thanks for the clarification!